src/mobu/services/business/notebookrunner.py

```diff
@@ -101,14 +101,30 @@ async def refresh(self) -> None:
     async def clone_repo(self) -> None:
         url = self.options.repo_url
         branch = self.options.repo_branch
+        ref = self.options.repo_ref
         with self.timings.start("clone_repo"):
-            await self._git.clone("-b", branch, url, str(self._repo_dir))
+            if ref:
+                await self._git.clone(url, str(self._repo_dir))
+                self._git.repo = self._repo_dir
+                await self._git.checkout(ref)
+            else:
+                await self._git.clone("-b", branch, url, str(self._repo_dir))
```

Comment:
It's weird that one execution path here leaves `self._git.repo` set and the other one doesn't, and that makes me worry about bugs at some point later on when someone assumes one behavior or the other. If one path sets `self._git.repo`, I think they both should.